database/insertData.js

```diff
@@ -81,10 +71,17 @@ const queries = [
                         options: ["Company A", "Company B"]
```

Comment:
I agree that we need to prevent this from happening. And for that we might need some `templateTypeChecking` for now since we will be entering most of them manually using GraphQL and only after the Template Builder is in place we can move to use the same `templateTypeChecking` for testing purposes. So, IMHO I would like to have the evaluateExpression making this check on front-end elements. Which for now would be only required to check the insert that we would be creating for new templates using the GraphQL. Basically it would scan the whole `gql` finding any type definition that should be strictly defined in the `Query Syntax` and warn for problems. What do you think? Does that cover this problem? I'll also update this file with your suggestion. : )

Reply:
Actually, what I think might be best is to add an additional check to `evaluateExpression`: if the input data doesn't conform to the correct node structure, it simply returns the input object unmodified. That way, we can handle values that aren't expressed in query format.

I've already got it setup to handle JSON strings (vs actual objects), so it would be fairly straightforward to add this additional check.

You think that's a reasonable solution?

Also, regarding checking queries -- I'm thinking of making a little React-based GUI for testing expressions, kind of like the Graphical query builder for PostGraphile (but less complex obviously). It's probably a little off-track for the time being, so I might try and squeeze it in as a little weekend project sometime.

Reply:
I think this would be covered by the issue: #58. We had a meeting on Friday (02/10/2020) that we discussed more the idea about this validation. If you have ideas to add to that issue it's better so we keep this discussion where it belongs. I'll resolve this comment now.